### PR TITLE
Add hillshade and contours for  peak plots 

### PR DESCRIPTION
**Pull request by awirb**

### CODE REVIEW

File: avaframe/out3Plot/plotUtils.py [hunk 1]
```diff
@@ -490,6 +501,42 @@ def putAvaNameOnPlot(ax, avaDir):
     return infoText
 
 
+def putInfoBox(ax, infoText, location='lowerRight', color='black', hAlignment='right', alphaF=0.5):
+    '''
+    Puts the  in the lower right or upper left or upper right or lowerLeft corner of the given
```
Comment (matthiasto) [inline avaframe/out3Plot/plotUtils.py:506]: I assume you mean put the infoText

File: avaframe/out3Plot/plotUtils.py [hunk 2]
```diff
@@ -490,6 +501,42 @@ def putAvaNameOnPlot(ax, avaDir):
...
+    matplotlib axes
+
+    Parameters
+    ----------
```
Comment (matthiasto) [inline avaframe/out3Plot/plotUtils.py:510]: you are missing some parameters and the return
Dont you also need to return the ax?

File: avaframe/out1Peak/outPlotAllPeak.py [hunk 3]
```diff
@@ -8,11 +8,13 @@
 import numpy as np
 from matplotlib import pyplot as plt
 import pathlib
+from matplotlib.colors import LightSource
 
 import avaframe.out3Plot.plotUtils as pU
 import avaframe.in1Data.getInput as gI
 from avaframe.in3Utils import fileHandlerUtils as fU
 import avaframe.in2Trans.ascUtils as IOf
+import avaframe.out3Plot.outTopo as oP
```
Comment (matthiasto) [inline avaframe/out1Peak/outPlotAllPeak.py:17]: what is this for?

File: avaframe/out1Peak/outPlotAllPeak.py [hunk 4]
```diff
@@ -110,15 +112,28 @@ def plotAllPeakFields(avaDir, cfgFLAGS, modName, demData=''):
...
+            im0 = ax.imshow(ls.hillshade(demConstrained, vert_exag=pU.vertExag, dx=demConstrained.shape[1],
+                dy=demConstrained.shape[0]), cmap='gray', extent=[colsMinPlot, colsMaxPlot,
+                rowsMinPlot, rowsMaxPlot], origin='lower', aspect='equal', zorder=1)
+            X, Y = oP._setCoordinateGrid(colsMinPlot, rowsMinPlot, cellSize, demConstrained)
+            CS =  ax.contour(X, Y, demConstrained, colors=['gray'], levels=pU.hillshadeContLevs, alpha=1.,
+                linewidths=0.5, zorder=2)
+            ax.clabel(CS, CS.levels, inline=True, fontsize=8,zorder=3)
+            pU.putInfoBox(ax, '- elevation [m]', location='upperLeft', color='gray',
+                hAlignment='left', alphaF=1.0)
+
```
Comment (matthiasto) [inline avaframe/out1Peak/outPlotAllPeak.py:127]: could we wrap this into a function so that we can use it for other plots too?

**Review by matthiasto — CHANGES_REQUESTED** [review]

File: avaframe/out1Peak/outPlotAllPeak.py (hunk 3 above)
Comment (awirb) [reply avaframe/out1Peak/outPlotAllPeak.py:17]: to create the meshgrid for the contours

File: avaframe/out3Plot/plotUtils.py (hunk 2 above)
Comment (awirb) [reply avaframe/out3Plot/plotUtils.py:510]: no I don't think so , removed the infoText return

File: avaframe/out1Peak/outPlotAllPeak.py (hunk 4 above)
Comment (awirb) [reply avaframe/out1Peak/outPlotAllPeak.py:127]: made a function for this

File: avaframe/out1Peak/outPlotAllPeak.py [hunk 5]
```diff
@@ -110,15 +110,20 @@ def plotAllPeakFields(avaDir, cfgFLAGS, modName, demData=''):
             rowsMaxPlot = (rowsMax+1)*cellSize
             colsMinPlot = colsMin*cellSize
             colsMaxPlot = (colsMax+1)*cellSize
-            im0 = ax.imshow(demConstrained, cmap='Greys', extent=[colsMinPlot, colsMaxPlot, rowsMinPlot, rowsMaxPlot], origin='lower', aspect='equal')
-            im1 = ax.imshow(data, cmap=cmap, norm=norm, extent=[colsMinPlot, colsMaxPlot, rowsMinPlot, rowsMaxPlot], origin='lower', aspect='equal')
+
+            # add DEM hillshade with contour lines
+            ls, CS = pU.addHillShadeContours(ax, demConstrained, colsMinPlot, rowsMinPlot, cellSize,
+                extent=[colsMinPlot, colsMaxPlot, rowsMinPlot, rowsMaxPlot])
```
Comment (matthiasto) [inline avaframe/out1Peak/outPlotAllPeak.py:116]: feels wierd to pass twice the colsMinPlot, rowsMinPlot arguments. is 'extent' really optional? dont we always need it here?

Comment (fso42) [reply avaframe/out1Peak/outPlotAllPeak.py:116]: If (!) this is really needed, then make an `extent` variable which you can also then use in line 119

File: avaframe/out3Plot/plotUtils.py [hunk 6]
```diff
@@ -374,10 +386,11 @@ def saveAndOrPlot(pathDict, outFileName, fig):
     outPath = None
     if cfgFlags.getboolean('savePlot'):
         outFileName = outFileName.replace(".", "_")
-        outname = os.path.join(pathDict['pathResult'], outFileName)
-        outPath = outname + '.' + outputFormat
-        if not os.path.exists(os.path.dirname(outname)):
-            os.makedirs(os.path.dirname(outname))
+        outDir = pathlib.Path(pathDict['pathResult'])
+        outPath = outDir / (outFileName + '.' + outputFormat)
+        outname = outDir / outFileName
```
Comment (fso42) [inline avaframe/out3Plot/plotUtils.py:391]: Since this is touched -> `outName`

**Review by fso42 — CHANGES_REQUESTED** [review]

**Review by fso42 — APPROVED** [review]

**Review by matthiasto — APPROVED** [review]

**Pull request merged**